src/js/jsx/shared/Datalist.jsx

```diff
@@ -410,8 +410,13 @@ define(function (require, exports, module) {
                     }) : null;
                 }
 
-                var suggestionID = suggestion ? suggestion.id : null,
+                var suggestionID = suggestion ? suggestion.id : this.state.id,
                     suggestionTitle = suggestion ? suggestion.title : "";
+
+                // If all the options are headers (no confirmable options, then set selected ID to null
+                if (!suggestion && collection.uniformValue(collection.pluck(options, "type"))) {
+                    suggestionID = null;
```

Comment:
Yeah, in this case ID should be `null` instead of staying the same because it means that there are no options in the dropdown. If it doesn't get set to `null`, then when you hit enter when there are no options, it will select the last valid option, which is not good.

My earlier attempt to fix this bug caused the issue that this PR fixes (selecting filters sometimes not working).